app/src/main/java/net/gsantner/markor/security/JavaPasswordbasedCryption.java

```diff
@@ -0,0 +1,233 @@
+     * @throws EncryptionFailedException when something goes wrong.
+     */
+    public byte[] encrypt(String plainText, char[] password) throws EncryptionFailedException {
+        return encryptBytes(plainText.getBytes(StandardCharsets.UTF_8), password);
+    }
+
+    /**
+     * Encrypt the given byte-array with the password.
+     *
+     * @param decryptedBytes the decrypted text.
+     * @param password       the password to create the key <b>Warning!</b> the array will be filled with 0!
+     * @return encrypted-bytes.
+     * @throws EncryptionFailedException when something goes wrong.
+     */
+    private byte[] encryptBytes(byte[] decryptedBytes, char[] password) throws EncryptionFailedException {
+        try {
+            final byte[] salt = getRandomBytes(version.keySaltLength);
+            final byte[] nonce = getRandomBytes(version.nonceLenth);
+            final SecretKey key = createKeyFromPassword(password, salt);
+            final byte[] cryptedBytes = getCipher(key, Cipher.ENCRYPT_MODE, nonce).doFinal(decryptedBytes);
+            final byte[] result = new byte[Version.NAME_LENGTH + nonce.length + salt.length + cryptedBytes.length];
+            System.arraycopy(version.name().getBytes(StandardCharsets.US_ASCII), 0, result, 0, Version.NAME_LENGTH);
+            System.arraycopy(nonce, 0, result, Version.NAME_LENGTH, version.nonceLenth);
+            System.arraycopy(salt, 0, result, Version.NAME_LENGTH + version.nonceLenth, version.keySaltLength);
+            System.arraycopy(cryptedBytes, 0, result, Version.NAME_LENGTH + version.nonceLenth + version.keySaltLength, cryptedBytes.length);
+            return result;
+        } catch (Exception e) {
+            throw new EncryptionFailedException("Can't encrypt text.", e);
+        }
+    }
+
+
+    /**
+     * Decrypt the given bytes with the password.
+     *
+     * @param encryptedText encrypted-bytes with version
+     * @param password      the password to create the key. <b>Warning!</b> the array will be filled with 0!!
+     * @return the decrypted text.
+     * @throws EncryptionFailedException when something goes wrong.
+     */
+    public String decrypt(byte[] encryptedText, char[] password) throws EncryptionFailedException {
+        byte[] decryptedCipherTextBytes = decryptBytes(encryptedText, password);
+        return new String(decryptedCipherTextBytes, StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Decrypt the given bytes with the password.
+     *
+     * @param encrypted encrypted-bytes with version
+     * @param password  the password to create the key <b>Warning!</b> the array will be filled with 0!
+     * @return the decrypted bytes.
+     * @throws EncryptionFailedException when something goes wrong.
+     */
+    private byte[] decryptBytes(byte[] encrypted, char[] password) throws EncryptionFailedException {
+        try {
+            final Version currentVersion = getVersion(encrypted);
+            if (currentVersion != version) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5279643)

Reply:
useless for enums but fixed.